src/plugins/console/README.md

```diff
@@ -8,11 +8,213 @@ Console provides the user with tools for storing and executing requests against
+Autocomplete definitions are all created in the form of javascript objects loaded from `json` and `js` files. 
+
+### Creating definitions
+The [`generated`](https://github.com/elastic/kibana/blob/master/src/plugins/console/server/lib/spec_definitions/json/generated) folder contains definitions created automatically from Elasticsearch REST API specifications. See this [README](https://github.com/elastic/kibana/blob/master/packages/kbn-spec-to-console/README.md) file for more information on the `spec-to-console` script. 
+
+Manually created override files in the [`overrides`](https://github.com/elastic/kibana/blob/master/src/plugins/console/server/lib/spec_definitions/json/overrides) folder contain fixes for generated files and additions for request body parameters.   
+
+### Top level keys
+Use following top level keys in the definitions objects.
+
+#### `documentation`
+Url to Elasticsearch REST API documentation for the endpoint (If the url contains `master` or `current` strings in the path, Console automatically replaces them with the `docLinkVersion` to always redirect the user to the correct version of the documentation).
+
+#### `methods`
+Allowed http methods (`GET`, `POST` etc)
+
+#### `patterns`
+Array of API endpoints (accepts variables like `{indices}` or `{fields}`, for example `{indices}/_rollup/{rollup_index}`, see below for more information)
+
+#### `url_params`
+Query url parameters and their values (see below for more information). An example: 
+```
+"url_params": {
+  "format": "",
+  "local": "__flag__",
+  "h": [],
+  "expand_wildcards": [
+    "open",
+    "closed",
+    "hidden",
+    "none",
+    "all"
+  ]
+}
+```
+
+#### `priority`
+Value for selecting one autocomplete definition, if several configurations are loaded from the files.
+
+#### `data_autocomplete_rules`
+Request body parameters and their values. Only used in `overrides` files because REST API specs don't contain any information about body request parameters.
+Refer to Elasticsearch REST API documentation when configuring this object. (see below for more information). An example:
+```
+"data_autocomplete_rules": {
+  "text": [],
+  "field": "{field}",
+  "analyzer": "",
+  "explain": { "__one_of": [false, true] },
+}
+```
+
+### Query url parameters
+Query url parameters are configured in form of an object, for example: 
+```
+"url_params": {
+  "local": "__flag__",
+  "scroll": "",
+  "expand_wildcards": [
+    "open",
+    "closed",
+    "hidden",
+    "none",
+    "all"
+  ]
+}  
+```
+This object specifies 3 query parameters `local` (boolean value), `scroll` (no default value) and `expand_wildcards` (with a list of accepted values).  
+When the user types in the url path into Console and at least 2 characters after `?`, all matching url parameters are displayed as autocomplete suggestions. In this example, after typing 
```

Comment:
Is this intended to be a new paragraph? If so can we add another line break, since it's rendering a bit funny:

![image](https://user-images.githubusercontent.com/1238659/132560911-bf262e47-ed3f-49d8-9750-5f717bb6bd48.png)